File: express/src/lib/jwt.ts
import jwt from "jsonwebtoken";
import { env } from "../config/env";

export function signToken(userId: string): string {
  return jwt.sign({ sub: userId }, env.JWT_SECRET, { expiresIn: env.JWT_EXPIRES_IN as `${number}${"s"|"m"|"h"|"d"|"w"|"y"}` });
}

export function verifyToken(token: string): { sub: string } {
  return jwt.verify(token, env.JWT_SECRET) as { sub: string };
}

File: fastapi/app/lib/jwt.py
from datetime import datetime, timedelta, timezone

import jwt

from ..config import settings


def sign_token(user_id: str) -> str:
    payload = {
        'sub': user_id,
        'iat': datetime.now(tz=timezone.utc),
        'exp': datetime.now(tz=timezone.utc) + timedelta(days=7),
    }
    return jwt.encode(payload, settings.JWT_SECRET, algorithm='HS256')


def verify_token(token: str) -> dict:
    return jwt.decode(token, settings.JWT_SECRET, algorithms=['HS256'])
